src/main/java/Duke.java

```diff
@@ -1,10 +1,177 @@
+import java.util.*;
+import java.io.*;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private static final String FILE_PATH = "./data/duke.txt";
+    public static void main(String[] args) throws Exception {
+        System.out.println("    ____________________________________________________________");
+        System.out.println("    Hello! I'm Yappy\n    What can I do for you?");
+        System.out.println("    ____________________________________________________________");
+
+        ArrayList<Task> AL = new ArrayList<>();
+
+        loadTasksFromFile(AL, FILE_PATH);
+
+        while(true) {
+            BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+            String input = br.readLine();
+            String[] token = input.split(" ");
+
+            if (input.equals("bye")) {
+                break;
+            } else if (token.length == 1 && !input.equals("list")) {
+                throw new DukeException("SOMETHING WENT WRONG!! Invalid input.");
+            } else if (input.equals("list")) {
+                System.out.println("    ____________________________________________________________");
+                System.out.println("     Here are the tasks in your list");
+                for (int i = 1; i <= AL.size(); i++) {
+                    System.out.println("       " + i + "." + AL.get(i - 1));
+                }
+                System.out.println("    ____________________________________________________________");
+            } else if (token[0].equals("mark")) {
+                AL.get(Integer.parseInt(token[1]) - 1).markAsDone();
+                System.out.println("    ____________________________________________________________");
+                System.out.println("     Nice! I've marked this task as done:");
+                System.out.println("       " + AL.get(Integer.parseInt(token[1]) - 1));
+                System.out.println("    ____________________________________________________________");
+            } else if (token[0].equals("unmark")) {
+                AL.get(Integer.parseInt(token[1]) - 1).markAsUndone();
+                System.out.println("    ____________________________________________________________");
+                System.out.println("     OK, I've marked this task as not done yet:");
+                System.out.println("       " + AL.get(Integer.parseInt(token[1]) - 1));
+                System.out.println("    ____________________________________________________________");
+            } else if (token[0].equals("deadline")) {
+//                Task[] tasks = new Task[100];
```

Comment:
This comment could be removed since you are already using an arraylist to store your tasks :)